### ISSUE TITLE
Version 2.0.2 SSH key problems on Windows 10

### ISSUE THREAD

**Issue by Jimmer-Ball** [hashicorp/vagrant#9511]
Issue with vagrant ssh on WIndows 10 with version 2.0.2 against latest VirtualBox (Version 5.2.6 r120293 (Qt5.6.2))

### Vagrant version
2.0.2

### Host operating system
Windows 10 Pro, version 1709, OS build 16299.2.48 and VirtualBox Version 5.2.6 r120293 (Qt5.6.2)

### Guest operating system
I tried a few out to be honest, to see if it was me or Vagrant before submitting the bug. Here are the ones I used

- Hashicorp/precise64
- debian/jessie64
- bento/ubuntu-16.04

Note replace simple box def in the Vagrantfile below with any of the above

### Vagrantfile
```ruby
# Try this out with all three flavours of base box if you want 
authentication tokens, or email addresses)
grant.configure(2) do |config|
  config.vm.hostname = "Supporting-Services"
  config.vm.provider :virtualbox do |vb|
  config.vm.box = "hashicorp/precise64"
  vb.customize ["modifyvm", :id,
            "--name", "Supporting Services",
            "--memory", "4096",
			"--cpus", "4",
            "--natdnshostresolver1", "on"]
  end
end
```

### Debug output
I don't have any debug output, the problem is a bit more general than that.

### Expected behavior
No password should need to be provided with vagrant ssh, it should work as it ever did

### Actual behavior
vagrant ssh results in prompt for password
...
password:?

Testing whether things are good, I also have cygwin shell (and ssh) installed on this box.

```
cd to where the key is
ssh -i the_private_key vagrant@127.0.0.1 -p2222 
```
works fine, no password required and straight in as expected

So, the vagrant ssh command is not picking up the right key file on Windows 10 with version 2.0.2

### Steps to reproduce
1. Install version 2.0.2 on Windows
2. Run Vagrantfile
3. See problem

Please note I already have a work-around, which is to use the 2.0.1 version of Vagrant, see below

### Work-around
1. Install version 2.0.1 on Windows 10
2. Run Vagrantfile
3. No problem with vagrant ssh, no password required as expected.

**Comment by oussaka** [comment]
I lost a lot of time because of this problem.
The solution for me was to install Virtualbox version 5.1.34 with vagrant 2.0.2.

**Comment by chrisroberts** [comment]
I believe this was an issue with permissions on the key file and the enforcement by the win32-openssh process. @Jimmer-Ball do you still experience this issue with the 2.0.3 release?

**Comment by WheelmanK** [comment]
I have the same issue.
I'm running Vagrant 2.0.3, VMware Workstation 14.1.1, the VMware desktop provider 1.0.0 plugin and the supporting VMware Utility program.  The host computer is running Windows 10 Pro with all current updates applied.
I ran the vagrant ssh command in debug mode and captured the output from which I extracted the complete command line that executes Vagrant's built in ssh program.  I then ran that in Powershell but removed the LogLevel=FATAL option so I could see the error being produced.
SSH reported the following:
### Warning: Permanently added 'www.xxx.yyy.zzz' (ECDSA) to the list of known hosts.
@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@
Permissions for './.vagrant/machines/default/vmware_desktop/private_key' are too open.
It is required that your private key files are NOT accessible by others.
This private key will be ignored.
Load key "./.vagrant/machines/default/vmware_desktop/private_key": bad permissions
vagrant@www.xxx.yyy.zzz's password:

The permissions set on the file are as follows:
Authenticated Users: Modify, Read & Execute, Read, Write
SYSTEM: Full control
Administrators: Full control
Users: Read & Execute, Read

I can run the exact same setup on a different host and am not required to enter a password.
The only difference is I'm directly logged into the host that works and I'm accessing the host that doesn't work via RDP.

**Comment by WheelmanK** [comment]
After posting the previous comment I continued to research what the differences may be between the 2 hosts I have that behave differently.  
I had collected debug info on the one that didn't work but not the one that did.
When I collected the debug info on the working host I noticed the path to the version of ssh.exe being executed was different. 
The working host executed c:\hashicorp\vagrant\embedded\usr\bin\ssh.exe
The non-working host executed c:\hashicorp\vagrant\embedded\bin\ssh.exe 

To determine if the version being executed mattered I renamed the instance found in c:\hashicorp\vagrant\embedded\bin on the non-working host and then issued "vagrant ssh" from the powershell command line.  
This time the expected behavior, no password prompt and direct login to the VM as the vagrant user was observed.

My conclusion is the version of the ssh tools found in the c:\hashicorp\vagrant\embedded\bin directory are the root of this problem.

BTW: The version of vagrant on both machines is 2.0.3 and the c:\hashicorp\vagrant\embedded\bin directory on the working host is empty.
